stock_reception_screen/models/stock_picking.py

```diff
@@ -41,3 +41,23 @@ def action_reception_screen_open(self):
                 "no_breadcrumbs": True,
             },
         }
+
+    def action_cancel(self):
+        # As the moves are posted individually through the reception screen, we
+        # have to put them in a backorder to cancel the remaining ones
+        for picking in self:
+            if picking.picking_type_code == "incoming":
+                done_moves = self.move_lines.filtered(lambda m: m.state == "done")
+                new_picking = self.copy({"move_lines": [], "backorder_id": picking.id})
+                # Set move lines after to bypass checks on creation made by
+                # the `_set_scheduled_date` method
+                new_picking.move_lines = done_moves
+                picking.message_post(
+                    body=_(
+                        'The backorder <a href="#" '
+                        'data-oe-model="stock.picking" '
+                        'data-oe-id="%d">%s</a> has been created.'
+                    )
+                    % (new_picking.id, new_picking.name)
+                )
```

Comment:
It would be nice to extract this in another method, so we are not stuck if we need to override this part.

Reply:
done